src/ImageLoaders/Pef/PEFContainer.cs

```diff
@@ -0,0 +1,135 @@
+#region License
+/* 
+ * Copyright (C) 2018-2021 Stefano Moioli <smxdev4@gmail.com>.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+#endregion
+using Reko.Core;
+using Reko.Core.Memory;
+using Reko.Core.Types;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Reko.ImageLoaders.Pef
+{
+    public class PefContainer
+    {
+        public readonly PEFContainerHeader ContainerHeader;
+        private readonly PEFSectionHeader[] sectionHeaders;
+        private readonly string?[] sectionNameTable;
+
+        public PefContainer(PEFContainerHeader hdr, PEFSectionHeader[] secHdrs, string[] sectionNameTable)
+        {
+            this.ContainerHeader = hdr;
+            this.sectionHeaders = secHdrs;
+            this.sectionNameTable = sectionNameTable;
+        }
+
+        /////
+
+        public IEnumerable<PefImageSegment> GetImageSegments(EndianByteImageReader rdr, Address addrLoad)
+        {
+            var addr = addrLoad;
+            for (int i=0; i<ContainerHeader.sectionCount; i++)
+            {
+                PEFSectionHeader sectionHeader = sectionHeaders[i];
+                
+                byte[] containerData = rdr.ReadAt(sectionHeader.containerOffset, rdr => rdr.ReadBytes(sectionHeader.packedSize));
+
+                if (sectionHeader.IsCompressedSection())
+                {
+                    MemoryStream output = new MemoryStream();
```

Comment:
I would probably move the body of the `if` statement to a method called `DecompressSection` for clarity.